### DISCUSSION TITLE
How to change `%uri=` value in `sqitch.plan`?

### DISCUSSION THREAD

**Discussion by halostatue** [theory/sqitch#801] [Q&A] [answered]
Years ago when we started using Sqitch, we were on bitbucket. We have since moved to GitHub and had never updated the `%uri=` field in `sqitch.plan`. If I update this, I get the following error:

```console
$ sqitch deploy
Cannot find change 105f459e22b030bda73880348920d65bb40c4fd7 (most_recent_change) in sqitch.plan
```

Is there a way to update the `%uri=` without this breaking?

**Comment by theory** [comment, marked as answer]
Yeas, this is because the URI is included in the data used to generate the SHA1 hash. Run `sqitch show change [change]` to see what's included in that info. An example from one of the tutorials:

``` sh
❯ sqitch show change @HEAD
project flipr
uri https://github.com/theory/sqitch-mysql-intro/
change change_pass
parent e993d9073f8bad4d385019b6e11554fed21fbe3a
planner Marge N. O’Vera <marge@example.com>
date 2013-12-31T22:12:39Z
requires
  + change_pass@v1.0.0-dev2

Change change_pass to use encyrpt().
```

I order to change it, you would have to rewrite every ID of every deployment --- not just changes, but also tags! That would require something like this:

1. With the old URI in the `%uri` pragma, deploy to a fresh new database
2. Change the `%uri` pragma in `sqitch.plan` and then `sqitch deploy` to a second new database
3. Map the new SHA1 IDs for changes and tags from the old registry to the new
4. Write a script to replace all the old IDs with the new ones in each table in the registry

Lotta work for no real advantage, since that URI is otherwise referenced very much, and is designed simply to allow one to have two projects with the same name but different URIs.

Another approach, if you don't care about losing the history of previous deployments, is to drop the entire registry schema (`DROP SCHEMA sqitch CASCADE`), set the new `%uri` pragma value, then deploy to the same change with the `--log-only` option. That will record al the changes as successfully deployed without actually deploying them.

**Reply by halostatue** [reply]
This is the answer. I don't much like it, but it is the answer. It would be nice to have a way to update this in the future.

**Reply by theory** [reply]
If you don't have many Sqitch projects and don't expect to run into duplicate names, you can just omit the URI when you create new projects.
